src/test/java/com/marklogic/kafka/connect/sink/MarkLogicSinkConnectorConfigBuilder.java

```diff
@@ -0,0 +1,75 @@
+package com.marklogic.kafka.connect.sink;
+
+import org.apache.kafka.connect.runtime.ConnectorConfig;
+
+import java.util.Properties;
+import java.util.UUID;
+
+public class MarkLogicSinkConnectorConfigBuilder {
+
+    private String topic = String.format("topic-%s", UUID.randomUUID().toString());
+
+    private String key = String.format("key-%s", UUID.randomUUID().toString());
+
+    private final Properties connectorProps = new Properties();
+
+    public MarkLogicSinkConnectorConfigBuilder withTopic(final String topic) {
+        this.topic = topic;
+        return this;
+    }
+
+    public MarkLogicSinkConnectorConfigBuilder withKey(final String key) {
+        this.key = key;
+        return this;
+    }
+
+    public <T> MarkLogicSinkConnectorConfigBuilder with(final String propertyName, final T value) {
+        connectorProps.put(propertyName, value);
+        return this;
+    }
+
+    public <T> MarkLogicSinkConnectorConfigBuilder withAll(final Properties connectorProps) {
+        this.connectorProps.putAll(connectorProps);
+        return this;
+    }
+
+    private <T> void ifNonExisting(final String propertyName, final T value) {
+        if (connectorProps.get(propertyName) != null) return;
+        connectorProps.put(propertyName, value);
+    }
+
+    public Properties build() {
+
+        ifNonExisting(ConnectorConfig.NAME_CONFIG, "marklogic-sink");
+        ifNonExisting(ConnectorConfig.CONNECTOR_CLASS_CONFIG, "MarkLogicSinkConnector");
+        ifNonExisting(ConnectorConfig.TASKS_MAX_CONFIG, "1");
+        ifNonExisting("topics", topic);
+        ifNonExisting("key", key);
+
+        ifNonExisting("ml.connection.host", "localhost");
+        ifNonExisting("ml.connection.port", "8000");
+        ifNonExisting("ml.connection.securityContextType", "DIGEST");
+        ifNonExisting("ml.connection.username", "admin");
+        ifNonExisting("ml.connection.password", "admin");
+        ifNonExisting("ml.document.format", "JSON");
+        ifNonExisting("ml.document.collections", "kafka-data");
+        ifNonExisting("ml.document.permissions", "rest-reader,read,rest-writer,update");
+        ifNonExisting("ml.document.uriPrefix", "/kafka-data/");
+        ifNonExisting("ml.document.uriSuffix", ".json");
+        ifNonExisting("ml.dmsdk.batchSize", "100");
+        ifNonExisting("ml.dmsdk.threadCount", "8");
+        ifNonExisting("key.converter", "org.apache.kafka.connect.storage.StringConverter");
```

Comment:
There are a bunch of warnings like this when the test is run:

    WARN  o.a.k.c.admin.AdminClientConfig - The configuration 'key.converter' was supplied but isn't a known config.

Do you know if those can be ignored? If so, I think it'd be good to add this to logback.xml:

```
<logger name="org.apache.kafka.clients" level="ERROR" additivity="false">
    <appender-ref ref="STDOUT"/>
  </logger>
```

That'll remove all the warnings about unrecognized configs, as the warnings are coming from multiple classes.

Reply:
How about if I do this in logback.xml?
  <root level="WARN">
    <appender-ref ref="STDOUT"/>
  </root>
  <logger name="com.marklogic" level="INFO" additivity="false">
    <appender-ref ref="STDOUT"/>
  </logger>
  <logger name="org.reflections" level="ERROR" additivity="false">
    <appender-ref ref="STDOUT"/>
  </logger>
  <logger name="org.apache.kafka.clients" level="ERROR" additivity="false">
    <appender-ref ref="STDOUT"/>
  </logger>

That gets rid of nearly all the non-marklogic logging, so it's much easier to see what's happening. If a bug requires debugging, the developer can always change the setting to get more information.

Reply:
That sounds good, and right, developer can always change the log settings temporarily.